Updating upgrade your widget guide (#3497)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/oie-upgrade-sign-in-widget/main/index.md

  baseUrl: 'https://{yourOktaDomain}',
  registration: {
    preSubmit: (postData, onSuccess, onFailure) => {
        // handle preSubmit callback
        onSuccess(postData);
    },
    postSubmit: (response, onSuccess, onFailure) => {
        // handle postsubmit callback
        onSuccess(response);
    }
  },
  features: {
    // Used to enable registration feature on the widget
    // https://github.com/okta/okta-signin-widget#feature-flags
    registration: true
  }
})
```

> **Note**: We have not included the `parseSchema` method in the above `registration` object; the ability to include additional schemas is no longer supported.

### IdP Discovery

IdP Discovery enables you to route users to different third-party IdPs that are connected to your Okta org. Users can federate back into the primary org after authenticating at the IdP. This feature still functions, but you no longer need to enable the link for users to initialize the route. Instead, you can configure a routing rule with the application context.

Remove the `idpDiscovery` property shown below:

```JavaScript
features: {
  idpDiscovery: true
}
```

### OpenID Connect/social authentication

> You no longer require the `idps` JavaScript object and `className` property in the widget (shown below for reference), and you can remove them.  This is optional as the Okta Identity Engine will now include IdPs based on the routing rules.

Reviewer: ```suggestion
You no longer require the `idps` JavaScript object and `className` property in the widget. You can remove the `idps` definition that is shown in the following snippet. 
```

Author: Agreed with your wish to change this, but I did something slightly different, which I'll send up here in the next commit.

```JavaScript
idps: [
  { type: 'GOOGLE', id: '0oagjkh' },
  { type: 'FACEBOOK', id: '0oagjkh' },
]

> ```

Reviewer: Add this after the snippet?

The widget displays IdPs based on Identity Engine routing rules.

Author: Again, pretty much did what you suggested just slightly differently. Really appreciate the idea.

### Smart card IdP

> You no longer need to configure the authentication settings for the [smart card IdP](https://help.okta.com/okta_help.htm?id=ext-idp-smart-card-workflow), as it's no longer supported.

Reviewer: ```suggestion
[Smart card IdP](https://help.okta.com/okta_help.htm?id=ext-idp-smart-card-workflow) is no longer supported.
```

> Remove the authentication settings for the smart card IdP (`piv`) shown below:

Reviewer: ```suggestion
Remove the authentication settings for the smart card IdP (`piv`):
```

```JavaScript
piv: {
  certAuthUrl: '/your/cert/validation/endpoint',
  text: 'Authenticate with a Smart Card',
  className: 'custom-style',
  isCustomDomain: true,
}
```

### Bootstrapping from a recovery token

If you're initializing the widget with a recovery token, the `recoveryToken` setting appears, for example:

```JavaScript
recoveryToken: 'x0whAcR02i0leKtWMZVc'
```

The recovery token is dynamic and is automatically passed into the initialization of the widget. A value in the `recoveryToken` setting currently doesn't have any effect on widget function, though, the setting takes effect in the future.

### Okta dashboard or custom dashboard sign-in flow

For an Okta dashboard sign-in, you no longer need to configure a redirect to the Okta Identity Cloud, create an Okta session, and then open a URL specified in the widget.

> Remove the redirect configuration (`setCookieAndRedirect()`) line shown below:

Reviewer: ```suggestion
Remove the redirect configuration (`setCookieAndRedirect()`) line, shown in the following snippet:
```
